lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -364,6 +371,22 @@ bool FrontendInputsAndOutputs::forEachInputProducingSupplementaryOutput(
                             : hasInputs() ? fn(firstInput()) : false;
 }
 
+bool FrontendInputsAndOutputs::forEachInputNotProducingSupplementaryOutput(
+    llvm::function_ref<bool(const InputFile &)> fn) const {
+  if (hasPrimaryInputs())
+    return forEachNonPrimaryInput(fn);
+  // If no primary inputs, compiler is in whole-module-optimization mode, and
+  // only the first input can produce supplementary outputs, although all
+  // inputs may contribute.
```

Comment:
Argh. This is what I meant by the name being wrong. We want the behavior you had before, even though it's not the complement of `forEachInputProducingSupplementaryOutput`, because it would be Very Bad if we dropped diagnostics from every file but the first one in WMO mode.